api/v1beta1/validator.go

```diff
@@ -30,7 +47,121 @@ func validOcid(ocid string) bool {
 }
 
 // validShape is a simple pre-flight
-// we will let the serverside handle the more complex and compete validation
+// we will let the serverside handle the more complex and compete validation.
 func validShape(shape string) bool {
 	return len(shape) > 0
 }
+
+// validRegion test if the string can be a region.
+func validRegion(stringRegion string) bool {
+	if blankRegex.MatchString(stringRegion) || stringRegion == "" {
+		return false
+	}
+	return true
+}
+
+// validateClusterName validates ClusterName.
+func (c *OCICluster) validateClusterName() field.ErrorList {
```

Comment:
I dont think we should validate cluster name, that is a kubernetes artifact name, and I think we should allow whatever Kubernetes allows when you create. resource. So best leave it to kubernetes?

Reply:
I will pull this out since right now we support 255 characters and Kubernetes cluster names max is 253 characters